x-pack/plugins/infra/public/hooks/use_breadcrumbs.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ChromeBreadcrumb } from 'kibana/public';
+import { i18n } from '@kbn/i18n';
+import { MouseEvent, useEffect } from 'react';
+import { EuiBreadcrumb } from '@elastic/eui';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+
+function handleBreadcrumbClick(
+  breadcrumbs: ChromeBreadcrumb[],
+  navigateToHref?: (url: string) => Promise<void>
+) {
+  return breadcrumbs.map((bc) => ({
+    ...bc,
+    ...(bc.href
+      ? {
+          onClick: (event: MouseEvent) => {
+            if (navigateToHref && bc.href) {
+              event.preventDefault();
+              navigateToHref(bc.href);
+            }
+          },
+        }
+      : {}),
+  }));
+}
+
+type AppId = 'logs' | 'metrics';
+
+export const makeBaseBreadcrumb = (
+  path: string,
+  observabilityPath: string,
+  app: AppId
+): [EuiBreadcrumb, EuiBreadcrumb] => {
+  const appText =
+    app === 'logs'
+      ? i18n.translate('xpack.infra.header.logsTitle', {
+          defaultMessage: 'Logs',
+        })
+      : i18n.translate('xpack.infra.header.metricsTitle', {
+          defaultMessage: 'Metrics',
+        });
+  return [
+    {
+      text: i18n.translate('xpack.infra.header.observabilityTitle', {
+        defaultMessage: 'Observability',
+      }),
+      href: observabilityPath,
+    },
+    {
+      text: appText,
+      href: path,
+    },
+  ];
+};
+
+export const useBreadcrumbs = (extraCrumbs: ChromeBreadcrumb[], app: AppId) => {
```

Comment:
The Observability plugin [exports a `useBreadcrumbs` hook](https://github.com/elastic/kibana/blob/master/x-pack/plugins/observability/public/hooks/use_breadcrumbs.ts) which you might be able to use here.

Reply:
Cool, would be good to see if this could help us out (totally fine if the answer is no for now for some reason).

Reply:
@smith Yep true I thought about using the existing hook at the beginning, which was rendering just `Observability` as the base breadcrumb. Then the idea was to create a new useBreadcrumbs hook in the infra plugin with an extra app parameter with the plan to move it to the observability plugin later on. Uptime for example does something similar, they have their own useBreadcrumb hook which renders `Observability > Uptime` as their base breadcrumb.  Similarly other plugins have their own implementation for the breadcrumbs.

I will keep the implementation in the infra plugin for now with the goal to move it to the Observability plugin later on to be possibly reused by other plugins as well. This would require of course a unified implementation to address the use cases of multiple plugins, for example handling the params when clicking on a breadcrumb etc.